jaxrs-api/src/main/java/jakarta/ws/rs/ext/Part.java

```diff
@@ -0,0 +1,195 @@
+* available at https://www.eclipse.org/legal/efsl.php.
+*******************************************************************/
+package jakarta.ws.rs.ext;
+
+import java.io.InputStream;
+import java.util.Optional;
+
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+
+/**
+ * A {@code Part} is one part of a multipart entity. As defined in
+ * <a href="https://tools.ietf.org/html/rfc7578">RFC 7578</a>, a multipart
+ * request or response must have a content type of "multipart/form-data" with a
+ * {@code boundary} parameter indicating where one part ends the next may begin.
+ * <p>
+ * Multipart entities may be received in a resource method as a collection of
+ * parts (ex: {@code List<Part>}) or as a form parameter (ex:
+ * {@code @FormParam("part1Name") Part part1}).
+ * </p>
+ * <p>
+ * Likewise, a client may receive a
+ * multipart response by reading the returned entity as a collection of Parts
+ * (ex: {@code response.readEntity(new GenericType<List<Part>>() {})}).
+ * </p>
+ * <p>
+ * In order to send a multipart entity either as a client request or a response
+ * from a resource method, you may create the Lists using {@code Part.Builder}.
+ * For example:
+ * </p>
+ * 
+ * <pre>
+ * Client c = ClientBuilder.newClient();
+ * WebTarget target = c.target(someURL);
+ * List&lt;Part&gt; parts = Arrays.asList(Part.newBuilder("name1").fileName("file1.doc").entityStream(stream1).build(),
+ *         Part.newBuilder("name1").fileName("file2.doc").entityStream(stream2).build(),
+ *         Part.newBuilder("name1").fileName("file3.doc").entityStream(stream3).build());
+ * Entity entity = Entity.entity(parts, MediaType.MULTIPART_FORM_DATA);
+ * Response r = target.request().post(entity);
+ * </pre>
+ * 
+ * Note that when building a Part, the name and entityStream are required.
+ * Other properties such as headers, file name, and media type are optional.
+ * 
+ * @since 3.1
+ */
+public interface Part {
+
+    /**
+     * Creates a new {@code Part.Builder} instance.
+     * 
+     * @param partName name of the part to create within the multipart entity
+     * @return {@link Builder} for building new {@link Part} instances
+     */
+    static Builder newBuilder(String partName) {
+        return RuntimeDelegate.getInstance().createPartBuilder(partName);
+    }
+
+    /**
+     * Returns the name of this part within the multipart entity. This will be the 
+     * "name" attribute of the {@code Content-Disposition} header for this part.
+     * 
+     * @return the part name
+     */
+    String getName();
+
+    /**
+     * Returns the filename of this part. This will be the "filename" attribute of
+     * the {@code Content-Disposition} header for this part. A filename is not
+     * required in a part, so if a filename is not present it will return
+     * {@code Optional.empty()}
+     * 
+     * @return an {@code Optional<String>} indicating the filename if present
+     */
+    Optional<String> getFileName();
```

Comment:
Is using Optional a good idea? What is the difference between Optional.empty() and null?

Reply:
I prefer `Optional` because it aligns with the RFC that the file name attribute is not required. It's also "showing my support for modern Java." ;-)

In my example, I think it makes it more readable to detect when the file name attribute exists or not - i.e. `String filename = part.getFileName().orElse(part.getName() + ".txt");`

Reply:
Yes, `Optional<?>` would have been used earlier in the API had it been available.